buildroot-external/board/nabucasa/green/uboot-boot.ush

```diff
@@ -27,6 +27,30 @@ test -n "${MACHINE_ID}" || setenv BOOT_CONDITION "systemd.condition-first-boot=t
 # HassOS bootargs
 setenv bootargs_hassos "zram.enabled=1 zram.num_devices=3 rootwait systemd.machine_id=${MACHINE_ID} fsck.repair=yes ${BOOT_CONDITION}"
 
+# Check if button is held for device wipe
+setenv counter 0
+i2c dev 0
+while test ${counter} -lt 2; do
+  i2c read 0x20 0xf0.1 1 ${loadaddr}
+  setexpr PWRON_STS *${loadaddr} \& 0x80
```

Comment:
We could make this a U-Boot environment, which we can run, e.g. `run check_pwron_sts`. 

That way we could just nest tests, get rid of the somewhat awkward while while still avoiding a lot of duplication.

Reply:
I just ignored it when there was just one occurrence, but since I like the falling edge idea from the other command, it will actually be useful :+1: